src/operations/indexes.ts

```diff
@@ -196,7 +219,25 @@ export class CreateIndexesOperation<
     this.options = options ?? {};
     this.collectionName = collectionName;
 
-    this.indexes = indexes;
+    // Ensure we generate the correct name if the parameter is not set
+    const namedIndexes = [];
+    for (const userIndex of indexes) {
+      const index: Omit<IndexDescription, 'key'> & { key: Map<string, IndexDirection> } = {
+        ...userIndex,
+        // Ensure the key is a Map to preserve index key ordering
+        key: userIndex.key instanceof Map ? userIndex.key : new Map(Object.entries(userIndex.key))
+      };
+      if (index.name == null) {
+        // Ensure every index is named
+        const keys = [];
+        for (const [name, direction] of index.key) {
+          keys.push(`${name}_${direction}`);
+        }
+        index.name = keys.join('_');
```

Comment:
I'm pretty sure the loop above is equivalent to:
`index.name = Array.from(index.key).flat().join('_');`

Reply:
done!